src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/KinesisDataFetcher.java

```diff
@@ -113,13 +125,18 @@ public GetRecordsResult getResult() {
 
         @Override
         public GetRecordsResult accept() {
+            if (!isValidResult(result)) {
+                throw new SdkClientException("GetRecordsResult is not valid. NextShardIterator: " + result.getNextShardIterator()
+                                                     + ". ChildShards: " + result.getChildShards());
+            }
             nextIterator = result.getNextShardIterator();
             if (!CollectionUtils.isNullOrEmpty(result.getRecords())) {
                 lastKnownSequenceNumber = Iterables.getLast(result.getRecords()).getSequenceNumber();
             }
             if (nextIterator == null) {
                 LOG.info("Reached shard end: nextIterator is null in AdvancingResult.accept for shard " + shardId);
```

Comment:
Should we add the list of childShardIds to this log message for debugging?

Reply:
Yeah, that sounds good.